maxim/blocks/misc_gating.py

```diff
@@ -116,24 +112,22 @@ def apply(x):
         u, v = tf.split(x, 2, axis=-1)
 
         # Get grid MLP weights
-        gh, gw = grid_size
-        fh, fw = h // gh, w // gw
-        u = BlockImages()(u, patch_size=(fh, fw))
-        dim_u = K.int_shape(u)[-3]
+        ghu, gwu = grid_size
+        u, phu, pwu = TFBlockImagesByGrid()(u, grid_size=(ghu, gwu))
+        dim_u = ghu * gwu
```

Comment:
Explain the rationale in the comment.

Reply:
Also, advisable not to change the original variable names here and elsewhere.

Reply:
Variable names will be recovered on next push. Did it only for readability (since they get rewritten a few lines below)

If i understood correclty, you  are asking why we can substitute K.int_shape(u)[-3] for (gh * gw):

From [BlockImages()](https://github.com/sayakpaul/maxim-tf/pull/24/files/d55b45acb0ff8287445a1df09104fc7fb9238c67#diff-52bd614bb8d4ae5da8df6a1a882250282cc2c6dfef96e6c904f1a12e08c586a6L29), we have that the output dimension is "b (gh gw) (fh fw) c". Thus, since:
```dim_u = K.int_shape(u)[-3]```
dim_u = (gh * gw)

Reply:
Same [reason](https://github.com/sayakpaul/maxim-tf/pull/24/files/d55b45acb0ff8287445a1df09104fc7fb9238c67#r1050913260) why fh and fw are getting replaced by gh and gw here?

Reply:
Essentially, those transformations are the same:
```python
def same_operations(random_image, grid_size=(gh,gw)):
    b, h, w, c = random_image.shape
    image_blocked_by_grid = BlockByGrid(random_image, grid_size=(gh, gw))
    image_blocked_by_block = BlockByPatch(random_image, patch_size=(h // gh, w // gw)
    image_blocked_by_grid == image_blocked_by_block # this should be True.
```
we have this pseudo-code as a test [here](https://github.com/LeviVasconcelos/maxim-tf/blob/main/maxim/tests/test_block_operations.py#L76-L86).